raiden_contracts/contract_manager.py

```diff
@@ -142,20 +142,30 @@ def get_event_abi(self, contract_name: str, event_name: str) -> Dict:
         contract_abi = self.get_contract_abi(contract_name)
         return find_matching_event_abi(contract_abi, event_name)
 
-    def _checksum_contracts(self):
-        if self._contracts_source_dirs is None:
-            raise TypeError("Can't checksum when using precompiled contracts.")
-        checksums = []
-        for contracts_dir in self._contracts_source_dirs.values():
+    @staticmethod
+    def checksum_contracts(contracts_source_dirs: list) -> dict:
+        checksums = {}
+        for contracts_dir in contracts_source_dirs.values():
```

Comment:
The type annotation says `contracts_source_dirs` is a list, but the code seems to expect a dict.